cli/flox/src/commands/environment.rs

```diff
@@ -423,18 +424,47 @@ impl Activate {
         let now_active =
             UninitializedEnvironment::from_concrete_environment(&concrete_environment)?;
 
-        let mut environment = concrete_environment.into_dyn_environment();
+        let environment = concrete_environment.dyn_environment_ref_mut();
 
         // Don't spin in bashrcs and similar contexts
-        let activation_path = if !stdout().is_tty() && self.run_args.is_empty() {
-            environment.activation_path(&flox)?
+        let activation_path_result = if !stdout().is_tty() && self.run_args.is_empty() {
+            environment.activation_path(&flox)
         } else {
             Dialog {
                 message: &format!("Getting ready to use environment {now_active}..."),
                 help_message: None,
                 typed: Spinner::new(|| environment.activation_path(&flox)),
             }
-            .spin()?
+            .spin()
+        };
+
+        let activation_path = match activation_path_result {
+            Err(
+                err @ EnvironmentError2::Core(CoreEnvironmentError::LockedManifest(
+                    LockedManifestError::BuildEnv(CallPkgDbError::PkgDbError(PkgDbError {
+                        exit_code: 123,
+                        ..
+                    })),
+                )),
+            ) => {
+                let mut message = formatdoc! {"
+                    This environment is not yet compatible with your system ({system}).
+
+                    {err}",
+                    system = flox.system
+                };
```

Comment:
```suggestion
                let mut message = format!(
                    "This environment is not yet compatible with your system ({system}).",
                    system = flox.system
                };
```

don't we want to fomrat errors with ❌?

Reply:
We're moving it left of `ERROR` https://github.com/flox/flox/issues/849

Reply:
yes, but we currently do consistently add it at the calling side.
I'd keep it that way until we deal with formatting/logging specifically

Reply:
If you prefer we can leave it, but it's not consistent
https://github.com/flox/flox/blob/8a69e2d2ab41c1c659dbbb99103ce31bdb3c48fc/cli/flox/src/commands/environment.rs#L1461

Reply:
Oh I wasn't aware we/I removed it there